pkg/ha/keepalive.go

```diff
@@ -104,14 +106,17 @@ func KeepAlive(ctx context.Context, cli *clientv3.Client, workerName string, kee
 			}
 		case <-ctx.Done():
 			log.L().Info("ctx is canceled, keepalive will exit now")
-			ctx, cancel := context.WithTimeout(cli.Ctx(), etcdutil.DefaultRequestTimeout)
-			cli.Revoke(ctx, lease.ID)
-			cancel()
 			return nil
 		}
 	}
 }
 
+func revokeLease(ctx context.Context, cli *clientv3.Client, id clientv3.LeaseID) (*clientv3.LeaseRevokeResponse, error) {
+	ctx, cancel := context.WithTimeout(cli.Ctx(), etcdutil.DefaultRevokeLeaseTimeout)
```

Comment:
how about adding a comment to highlight _we must ensure `cli.Ctx()` not done when exiting_ (do not set `Context` when creating `cli` or do not cancel this `Context` or it's parent context).

Reply:
Addressed in https://github.com/pingcap/dm/pull/548/commits/6e69519b2d12e077049442d350f92c75ff44fc43